app/utils/routeHelpers.js

```diff
@@ -25,7 +25,10 @@ export function updateCollectionUrl(
   collection: Collection
 ): string {
   // Update url to match the current one
-  return oldUrl.replace(`/collection/${collection.id}`, collection.url);
```

Comment:
This approach was not updating the URL if it is `wrong-collection-r9QsabjdGv` to `correct-collection-r9QsabjdGv`.

Reply:
aha!